sound/soc/sof/intel/intel-audio.c

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause)
+//
+// This file is provided under a dual BSD/GPLv2 license.  When using or
+// redistributing this file, you may do so under either license.
+//
+// Copyright(c) 2019 Intel Corporation. All rights reserved.
+//
+// Authors: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
+
+/* Intel-specific SOF audio client code */
+#include <sound/sof/dai.h>
+#include "../sof-client.h"
+#include "../sof-priv.h"
+#include "../sof-audio.h"
+#include "shim.h"
+#include "intel-audio.h"
+
+int intel_register_audio_clients(struct snd_sof_dev *sdev)
+{
+	struct snd_sof_pdata *plat_data = sdev->pdata;
+	const struct sof_dev_desc *desc = plat_data->desc;
+	const struct snd_sof_audio_ops *audio_ops = desc->audio_ops;
+	char *client_name;
+	size_t size;
+	int count;
+	u32 type;
+	int i;
+
+	/* number of audio clients to register */
+	sdev->sof_num_clients = audio_ops->num_dai_drv_info;
+
+	size = sdev->sof_num_clients * sizeof(*sdev->sof_clients);
+
+	sdev->sof_clients = devm_kzalloc(sdev->dev, size, GFP_KERNEL);
+	if (!sdev->sof_clients)
+		return -ENOMEM;
+
+	/* register clients. Any of these can fail but keep going */
+	for (i = 0; i < audio_ops->num_dai_drv_info; i++) {
+		type = audio_ops->dai_drv_info[i].type;
+		count = snd_sof_get_dai_drv_count(audio_ops, type);
+
+		switch (type) {
+		case SOF_DAI_INTEL_SSP:
+			client_name = "sof-ssp-audio";
+			break;
+		case SOF_DAI_INTEL_DMIC:
+			client_name = "sof-dmic-audio";
+			break;
+		case SOF_DAI_INTEL_HDA:
+			client_name = "sof-hda-audio";
+			break;
+		}
+
+		sdev->sof_clients[i] =
+				sof_client_dev_register(sdev, client_name);
+		if (!sdev->sof_clients[i])
+			dev_warn(sdev->dev,
+				 "sof-ssp-audio client failed to register\n");
+	}
```

Comment:
Make this message generic, you can use client_name instead of "sof-ssp-audio"

Reply:
oops yes, thanks!